palettable/utils.py

```diff
@@ -190,8 +195,10 @@ def get_map(name, reverse=False):
         if name_lower not in name_map:
             raise KeyError('Unknown palette name: {0}'.format(name))
         name = name_map[name_lower]
-
-        colors = evenly_spaced_values(length, names_to_data[name])
+        if is_evenly_spaced:
+            colors = evenly_spaced_values(length, names_to_data[name])
+        else:
+            colors = names_to_data[name][0:length]
```

Comment:
It looks like this PR is engineered so this will never have `length > len(names_to_data[name])`, but I wonder if there should be a check for future usage? Also, if you want, you can leave the `0` off this slice and do `colors = names_to_data[name][:length]`.